docs/install-config/configure-internal-tls.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Configure Internal TLS communication between Harbor Component
+weight: 30
+---
+
+ By default, The internal communication between Harbor's component (harbor-core,harbor-jobservice,proxy,harbor-portal,registry,registryctl,trivy_adapter,clair_adapter,chartmuseum) use HTTP protocol which might not be secure enough for some production environment. Since Harbor v2.0, TLS can be used for this internal network. In production environments, always use HTTPS is a recommended best practice.
+
+This functionality is introduced via the `internal_tls` in `harbor.yml` file. To enabled internal TLS, set `enabled` to `true` and set the `dir` value to the path of directory that contains the internal cert files.
+
+All certs can be automatically generated by `prepare` tool.
+```bash
+docker run -v /:/hostfs goharbor/prepare:v2.0 gencert -p /path/to/internal/tls/cert
+```
+
+User also can provides their own CA to generate the other certs. Just put CA key pair on internal tls cert directory and name them as `harbor_internal_ca.key` and `harbor_internal_ca.crt`.
+Besides, User could provide all the certs by themselves either. However, there are some constraints for the certs:
```

Comment:
1. User -> user
2. I don't think `either` is correct, you should say "a user can also provide the certs for all components"